File: python_fastapi/src/crud.py
from sqlalchemy.orm import Session
from . import models, schemas


def create_task(db: Session, task: schemas.NewTask):
    db_task = models.Task(**task.model_dump())
    db.add(db_task)
    db.commit()
    return schemas.Task.model_validate(db_task)


def get_task(db: Session, task_id: str):
    db_task = db.query(models.Task).filter(models.Task.id == task_id).first()
    return schemas.Task.model_validate(db_task) if db_task else None


def get_tasks(db: Session):
    db_tasks = db.query(models.Task).all()
    return [schemas.Task.model_validate(task) for task in db_tasks]


def update_task(db: Session, task_id: str, task: schemas.NewTask):
    db_task = db.query(models.Task).filter(models.Task.id == task_id).first()
    if db_task:
        for key, value in task.model_dump().items():
            setattr(db_task, key, value)
        db.commit()
        return schemas.Task.model_validate(db_task)
    return None


def delete_task(db: Session, task_id: str):
    db_task = db.query(models.Task).filter(models.Task.id == task_id).first()
    if db_task:
        db.delete(db_task)
        db.commit()
        return 1
    return 0

File: rust_axum/src/crud.rs
use crate::models::{NewTask, Task};
use crate::schema::tasks;
use diesel::prelude::*;
use diesel::sqlite::SqliteConnection;

pub fn create_task(conn: &mut SqliteConnection, new_task: &NewTask) -> QueryResult<Task> {
    conn.transaction(|conn| {
        diesel::insert_into(tasks::table)
            .values(new_task)
            .execute(conn)?;

        tasks::table.order(tasks::id.desc()).first(conn)
    })
}

pub fn get_task(conn: &mut SqliteConnection, task_id: i32) -> QueryResult<Task> {
    tasks::table.find(task_id).first(conn)
}

pub fn get_tasks(conn: &mut SqliteConnection) -> QueryResult<Vec<Task>> {
    tasks::table.load::<Task>(conn)
}

pub fn delete_task(conn: &mut SqliteConnection, task_id: i32) -> QueryResult<usize> {
    diesel::delete(tasks::table.find(task_id)).execute(conn)
}

pub fn update_task(
    conn: &mut SqliteConnection,
    task_id: i32,
    updated_task: &NewTask,
) -> QueryResult<Task> {
    diesel::update(tasks::table.find(task_id))
        .set(updated_task)
        .execute(conn)?;

    tasks::table.find(task_id).first(conn)
}